### PR TITLE
-f option makes rm much easier :-)

### PR DESCRIPTION
**Pull request by EricManiez**

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -23,23 +23,27 @@ filenamesc := ft_strlen.c \
...
-test: test.c libft.a libft.h
-	$(CC) -o test libft.a test.c
+test: test.c $(NAME) libft.h
+	$(CC) -o test $(NAME) test.c
 
-libft.a: $(filenameso) 
-	ar -rc libft.a $(filenameso)
+$(NAME): $(filenameso) 
+	ar -rc $(NAME) $(filenameso)
+	ranlib $(NAME)
```
Comment (MarcManiez) [inline makefile:46]: Can you sell me on ranlib? What can I do with this that I couldn't before?

File: makefile [hunk 2]
```diff
@@ -49,7 +53,10 @@ dotos: $(filenameso)
 #=======#
 
 clean:
-	rm libft.a test $(filenameso)
+	rm -f $(NAME)
+	rm -f $(filenameso)
```
Comment (MarcManiez) [inline makefile:57]: What's the thinking behind writing this in multiple lines? #flyfucking

File: makefile [hunk 3]
```diff
@@ -23,23 +23,27 @@ filenamesc := ft_strlen.c \
 			ft_strncmp.c \
 			ft_atoi.c \
```
Comment (MarcManiez) [inline makefile:24]: Is placing this on multiple lines mostly an aesthetic choice? Either way I can dig it.

File: makefile [hunk 4]
```diff
@@ -23,23 +23,27 @@ filenamesc := ft_strlen.c \
...
 
-filenameso := $(filenamesc:%.c=%.o) 
-CC = gcc -Wall -Wextra -Werror
+filenameso := $(filenamesc:%.c=%.o)
+
+CC = gcc
+CFLAGS = -Wall -Wextra -Werror
+NAME = libft.a
```
Comment (MarcManiez) [inline makefile:30]: I'm not sure I get the advantage of not hardcoding it, here. Doesn't adding a variable just make things harder for that poor little computer of ours?

**Review by MarcManiez — CHANGES_REQUESTED** [review]
J'ai enculé toutes les mouches. En environnement pro, je crois que j'aurai tout laissé passer sans questions histoire d'aller vite, mais vu que j'essaie d'apprendre un max, j'ai posé toutes mes questions.

File: makefile (hunk 4 above)
Comment (EricManiez) [reply makefile:30]: It does, but it also makes it a lot easier for you to reuse your Makefile for something else afterwards.

File: makefile (hunk 3 above)
Comment (EricManiez) [reply makefile:24]: Aesthetic, and useful if there's one or more files you don't want to compile for some reason.
Much easier to comment out.

File: makefile (hunk 2 above)
Comment (EricManiez) [reply makefile:57]: Could do it in one line.
Or maybe you'll want to comment out only one of the instructions someday, and it'll be a lot faster to do so with 2 separate instructions !

File: makefile (hunk 1 above)
Comment (EricManiez) [reply makefile:46]: It's supposed to index your library to make it faster.
As you can see, I don't know too much about the subject.

**Review by MarcManiez — APPROVED** [review]
Let's do this.

**Pull request merged**